### ISSUE TITLE
Pro License Renewal seems to me missing a warning feature

### ISSUE THREAD

**Issue by jaysmithgit** [RealRaven2000/QuickFolders#164]
I had been using a Pro test license granted by Axel because we were working on a specific problem and Axel needed me to be testing.  (I had also purchased a separate Pro license testing on another seat/user.)  I had completely forgotten that I was still using a testing license.

I had to restart Thunderbird (and the whole workstation and the whole server and network) due to a major, extended power outage and related power problems (which overwhelmed our backup batteries, etc.).

!!! Normally I only restart Thunderbird (or my workstation) every few weeks or every few months -- only when there is some sort of Thunderbird problem or a major long-term power outage. 

When I restarted Thunderbird, there was suddenly a red button telling me that I needed to renew my QF Pro license.  It said that the license had expired 5 days previous!  But I had not known it had expired.

I suspect that this means that license / renewal status is only being checked when Thunderbird is started.

It would be much more convenient, if it is possible, ...

1) For the status to be checked every day even if Thunderbird is already/still running, not only if being restarted.

2) And, if an "about to expire" message/warning does not already exist (I don't know if one exists because my Thunderbird had not been restarted for weeks), then add an "about to expire" message that appears a couple of days before expiration so that the user has time to conveniently do the renewal.

3) If all of that is not possible / practical, then perhaps give the user 1-2 days of grace from the time that they are first shown the red button (at which point Pro features no longer function), so that the user can conveniently make the renewal purchase.  Since the Pro features (especially using the keyboard for filing) are so critical for daily use, it can be a problem if the user does not immediately have the necessary credit card number with them at that moment.

Thank you.

**Comment by fredstluka** [comment]
I agree.  More warning about expiration would be useful.
My Pro license expired the other day, and I got no notification
at all.  None in advance, and none after it happened.  The
Pro keyboard shortcuts just stopped working.

I had no idea what the problem was.  I assumed that all of QF
was failing, perhaps due to a recent TB upgrade or something.
So, I clicked on the QF cog to see the version number and to
check that the shortcut keys were still defined.  The options
page popped up saying in a large red block that I had expired.

Once I knew what was up, it was easy to renew, get the new
license key, and enter it.

Thanks!

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> I agree.  More warning about expiration would be useful. My Pro license expired the other day, and I got no notification at all.  None in advance, and none after it happened.  The Pro keyboard shortcuts just stopped working.

Did you not see the big red "Renew Quickfolders" button on the left hand side of the QuickFolders toolbar?

![image](https://user-images.githubusercontent.com/6314674/122196003-c0c8e000-ce8e-11eb-8ec8-2e741064c41b.png)

I think this would be visible for the majority of users (who use the toolbar mainly)

if it's about to expire, I agree that only showing an "Extend License" button on the QF Pro tab is too hidden:
![image](https://user-images.githubusercontent.com/6314674/122196278-02598b00-ce8f-11eb-9800-c556aa966a99.png)

**Comment by fredstluka** [comment]
> Did you not see the big red "Renew Quickfolders" button on the left 
> hand side of the QuickFolders toolbar?
>

No, I did not.  Is that the toolbar that sits below the list of messages
and above the preview area?

--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by RealRaven2000** [comment]
> > Did you not see the big red "Renew Quickfolders" button on the left hand side of the QuickFolders toolbar?

> No, I did not.  Is that the toolbar that sits below the list of messages and above the preview area?

No it's the main toolbar. There is a normally "uncolored" label that reads "QuickFolders" on the left, looks like this:
![image](https://user-images.githubusercontent.com/6314674/122385582-e7594a80-cf64-11eb-8c43-6683e3caeef7.png)

When hovering it turns green:
![image](https://user-images.githubusercontent.com/6314674/122385636-f6d89380-cf64-11eb-95f1-96abdfa70420.png)

However, it can be hidden via  Configuration / Advanced / Miscellaneus
![image](https://user-images.githubusercontent.com/6314674/122385763-17a0e900-cf65-11eb-88a3-48136ee00dd8.png)

But even if disabled, when the license has run out it will again be displayed with the text "Renew License" on Crimson background, like this:
![image](https://user-images.githubusercontent.com/6314674/122385979-4cad3b80-cf65-11eb-91c3-493abaa17627.png)

I assumed this would be a strong signal for clicking it to get to the license screen. Should it be displayed a few days before license expiry?

**Comment by fredstluka** [comment]
Axel,

Yes, showing a warning a few days before expiration would be
a good idea.  Or offering a few days "grace period".  As Jay said,
QuickFolders rapidly becomes a critical tool for Thunderbird users.
It would be a real nuisance to have to do without it on short notice,
especially at a time when I'm too busy to stop what I'm doing and
renew the license or don't have my credit card handy.

The red button you describe sounds perfect!  I agree it should be
very visible.  However, I didn't see it.  I suspect it wasn't displayed
in my case.

I DO have the full QuickFolders button displayed in the main Toolbar.
When the Pro keys stopped working, the 1st thing I thought to do was
to click that QuickFolders button.  If it had been replaced by a red
"Renew License" button, I would think I'd have noticed.

After I clicked it, and chose "QuickFolders Options..." I saw the usual
Options dialog but with one new bright red tab that said I was expired.
At least, I think that's how it went.  It was a week ago, so I don't
remember exactly.  In any case, I wasn't confused for long, and did
manage to renew quickly.

I don't remember how long I'd had Thunderbird open, or whether
I closed and re-opened it to get the Pro keys to work again.  Unlike
Jay, I do NOT typically leave it running for weeks or months at a
time.  I usually close it every evening before I run my daily backups,
and re-open it every morning.  And I sometimes close and re-open
it a couple times during the day.  So it hadn't been running for more
a day or 2 at most.

I'm a cautious user who's had lots of experience with apps that
leaked memory over time, or locked files against backups, etc.  So,
I tend to close any app that I'm not currently using.  Also, as I
mentioned in a separate ticket:
- 
https://github.com/RealRaven2000/quickfolders/issues/154#issuecomment-861911056
I had a problem for over a year with Thunderbird sometimes
deleting my .MSF files, and that tended to happen more often when
I'd had it open for a few hours.  So I still tend to close it and re-open
it now and then, just to reset things.

Is it possible that my case is different because of the fact that I've
not only paid for a Pro subscription, but also made a couple of
donations?  Did you do anything to my subscription based on my
donations?  Maybe I was only halfway expired?  Is there more than
one place in the code/data that records my expiration date and
status, that they could have been out of sync with each other?

Anyhow, it's all working fine now, and I've added a reminder to
my calendar for next year when it expires again, to watch for that
big red button and let you know whether it appears or not.

Thanks!
--Fred
------------------------------------------------------------------------
Fred Stluka -- http://bristle.com -- Glad to be of service!
Open Source: Without walls and fences, we need no Windows or Gates.
------------------------------------------------------------------------

**Comment by fredstluka** [comment]
> Did you not see the big red "Renew Quickfolders" button on the left hand side of the QuickFolders toolbar?

A year later, my license has expired agajn, and YES, this time I did see a big red button that made it obvious that I needed to renew.  Thanks!

It would still be a good idea to warn users a few days in advance, or to offer a grace period.  But, for me (with no corporate headaches of trying to get an employer to pay for the license), it was quick and easy to renew once it expired.

--Fred
